### PR TITLE
Sync ingested files only if reopen is supported by the FS

### PR DESCRIPTION
**Pull request by anand1976**
Some file systems (especially distributed FS) do not support reopening a file for writing. The ExternalSstFileIngestionJob calls ReopenWritableFile in order to sync the ingested file, which typically makes sense only on a local file system with a page cache (i.e Posix). So this change tries to sync the ingested file only if ReopenWritableFile doesn't return Status::NotSupported().

Tests:
Add a new unit test in external_sst_file_basic_test

### CODE REVIEW

**Review by jay-zhuang — APPROVED** [review]
LGTM.
Please add a note in History.md

File: db/external_sst_file_ingestion_job.cc [hunk 1]
```diff
@@ -109,17 +109,26 @@ Status ExternalSstFileIngestionJob::Prepare(
...
-                           "Failed to sync ingested file %s: %s",
-                           path_inside_db.c_str(), status.ToString().c_str());
+        Status s = fs_->ReopenWritableFile(path_inside_db, env_options_,
+                                           &file_to_sync, nullptr);
+        TEST_SYNC_POINT_CALLBACK("ExternalSstFileIngestionJob::Prepare:Reopen",
+                                 &s);
+        // Some file systems (especially remote/distributed) don't support
+        // reopening a file for writing and don't require reopening and
+        // syncing the file. Ignore the NotSupported error in that case.
+        if (!s.IsNotSupported()) {
```
Comment (jay-zhuang) [inline db/external_sst_file_ingestion_job.cc:119]: I'm wondering if the user should explicitly return a status for skip sync file, or ignore reopen error. As the default is `NotSupported`, should the user return a different status for that? or NotSupported with a special SubCode.

**Review by ajkr — COMMENTED** [review]
It looks like fsyncing ingested file data was bundled with the solution to #5287, but it feels extraneous and potentially misleading in what scenarios we can handle.

**Review by ajkr — APPROVED** [review]
I'd prefer delete the file data sync and document the expectation in `IngestExternalFile()`. But in any case, skipping this logic in `NotSupported` case seems like a step forward.

File: db/external_sst_file_ingestion_job.cc [hunk 2]
```diff
@@ -109,17 +109,26 @@ Status ExternalSstFileIngestionJob::Prepare(
         // directory before ingest the file. For integrity of RocksDB we need
```
[the remark this answers is not in the document]
Comment (anand1976) [reply db/external_sst_file_ingestion_job.cc:109]: Its probably easier for the app developer to forget to fsync the file than to forget to handle the fsync error. But yes, I think its reasonable to assume that the app should do whatever's necessary to persist the data before calling this.

**Pull request closed without merging**